pkg/scheduler/core/equivalence_cache.go

```diff
@@ -122,17 +123,14 @@ func (ec *EquivalenceCache) PredicateWithECache(
 		if cachePredicate, exist := algorithmCache.predicatesCache.Get(predicateKey); exist {
 			predicateMap := cachePredicate.(PredicateMap)
 			// TODO(resouer) Is it possible a race that cache failed to update immediately?
-			if hostPredicate, ok := predicateMap[equivalenceHash]; ok {
-				if hostPredicate.Fit {
-					return true, []algorithm.PredicateFailureReason{}, false
-				}
-				return false, hostPredicate.FailReasons, false
+			if hostPredicate, exist := predicateMap[equivalenceHash]; exist {
+				metrics.EquivalenceCacheHits.Inc()
+				return hostPredicate.Fit, hostPredicate.FailReasons, false
 			}
-			// is invalid
-			return false, []algorithm.PredicateFailureReason{}, true
 		}
 	}
-	return false, []algorithm.PredicateFailureReason{}, true
+	metrics.EquivalenceCacheMisses.Inc()
```

Comment:
I think `cachePredicate` does not exist should also be a case of ecache miss, as it indicates this record was invalidated sometime.

Reply:
I agree. The way I've changed this method, there is only one return statement for invalid=true, so we only need to update that metric here.